consensus/src/quorum_store/utils.rs

```diff
@@ -188,159 +215,162 @@ impl ProofQueue {
+        let mut remaining_txns = 0;
+        let mut remaining_proofs = 0;
+        let mut remaining_local_txns = 0;
+        let mut remaining_local_proofs = 0;
 
-        for (digest, expiration) in self.local_digest_queue.iter() {
-            // Not expired. It is possible that the proof entry in digest_proof was already removed
-            // when draining the digest_queue but local_digest_queue is not drained yet.
-            if *expiration >= current_block_timestamp {
-                if let Some(entry) = self.digest_proof.get(digest) {
+        // TODO: if the digest_queue is large, this may be too inefficient
+        let peers: Vec<_> = self.author_to_batches.keys().cloned().collect();
+        for peer in peers {
+            // TODO: queue direction!
+            let queue = self.author_to_batches.get(&peer).unwrap();
+            for batch in queue.iter() {
+                // Not expired
+                if batch.expiration() >= current_block_timestamp {
                     // Not committed
-                    if entry.is_some() {
-                        remaining_local_proof_size += 1;
+                    if let Some(Some((proof, _))) = self.batch_to_proof.get(batch) {
+                        remaining_txns += proof.num_txns();
+                        remaining_proofs += 1;
+                        if proof.author() == self.my_peer_id {
+                            remaining_local_txns += proof.num_txns();
+                            remaining_local_proofs += 1;
+                        }
                     }
                 }
             }
         }
-        counters::NUM_LOCAL_PROOFS_LEFT_ON_COMMIT.observe(remaining_local_proof_size as f64);
+        counters::NUM_TOTAL_TXNS_LEFT_ON_COMMIT.observe(remaining_txns);
+        counters::NUM_TOTAL_PROOFS_LEFT_ON_COMMIT.observe(remaining_proofs);
+        counters::NUM_LOCAL_TXNS_LEFT_ON_COMMIT.observe(remaining_local_txns);
+        counters::NUM_LOCAL_PROOFS_LEFT_ON_COMMIT.observe(remaining_local_proofs);
 
-        if let Some(&(_, time)) = self.local_digest_queue.iter().next() {
-            Some(time)
-        } else {
-            None
-        }
+        (remaining_txns, remaining_proofs)
     }
 
-    //mark in the hashmap committed PoS, but keep them until they expire
-    pub(crate) fn mark_committed(&mut self, digests: Vec<HashValue>) {
-        for digest in digests {
-            self.digest_proof.insert(digest, None);
-            if let Some(insertion_time) = self.digest_insertion_time.get(&digest) {
+    // Mark in the hashmap committed PoS, but keep them until they expire
+    pub(crate) fn mark_committed(&mut self, batches: Vec<BatchInfo>) {
+        for batch in batches {
+            if let Some(Some((_, insertion_time))) = self.batch_to_proof.get(&batch) {
                 counters::POS_TO_COMMIT.observe(insertion_time.elapsed().as_secs_f64());
             }
+            self.batch_to_proof.insert(batch, None);
```

Comment:
Why keep a record of the batch if it is committed? It would be simpler to remove it right?